src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -475,20 +475,24 @@ template <typename... AddMutableItemTags, typename AddImmutableItemTagsList,
 constexpr DataBox<tmpl::list<Tags...>>::DataBox(
     tmpl::list<AddMutableItemTags...> /*meta*/,
     AddImmutableItemTagsList /*meta*/, Args&&... args) noexcept {
-  DEBUG_STATIC_ASSERT(sizeof...(AddMutableItemTags) == sizeof...(Args),
-                      "Must pass in as many arguments as AddTags");
+  if constexpr (sizeof...(Args) > 0) {
+    DEBUG_STATIC_ASSERT(sizeof...(Args) == 0 or
+                            sizeof...(Args) == sizeof...(AddMutableItemTags),
+                        "Must pass in as many arguments as AddTags, or none to "
+                        "default-construct them.");
 #ifdef SPECTRE_DEBUG
-  // The check_argument_type call is very expensive compared to the majority of
-  // DataBox
-  expand_pack(detail::check_initialization_argument_type<
-              AddMutableItemTags, typename AddMutableItemTags::type,
-              std::decay_t<Args>>()...);
+    // The check_argument_type call is very expensive compared to the majority
+    // of DataBox
+    expand_pack(detail::check_initialization_argument_type<
+                AddMutableItemTags, typename AddMutableItemTags::type,
+                std::decay_t<Args>>()...);
 #endif  // SPECTRE_DEBUG
 
-  std::tuple<Args&&...> args_tuple(std::forward<Args>(args)...);
-  add_items_to_box(args_tuple, tmpl::list<AddMutableItemTags...>{},
-                   std::make_index_sequence<sizeof...(AddMutableItemTags)>{},
-                   AddImmutableItemTagsList{});
+    std::tuple<Args&&...> args_tuple(std::forward<Args>(args)...);
+    add_items_to_box(args_tuple, tmpl::list<AddMutableItemTags...>{},
```

Comment:
I don't think you want to skip this function completely.  You need to call the `EXPAND...` on line 451 otherwise you will bypass some static_assert checks.  (You may trigger other compiler errors, but they may not be as easy to parse)

Reply:
Fixed, and added more tests